packages/algob/src/lib/tx.ts

```diff
@@ -34,6 +35,21 @@ async function getSuggestedParamsWithUserDefaults (
   return suggested;
 }
 
+export async function getSuggestedParamsWithUserDefaultsASC (
+  algocl: tx.Algodv2, userDefaults: ASCPaymentFlags): Promise<tx.SuggestedParams> {
+  const suggested = await getSuggestedParams(algocl);
+  suggested.fee = userDefaults.totalFee === undefined
```

Comment:
Please reuse the method from `asa.ts`. As the fields are almost all the same you can use all of it or most of it.
If you don't need a `creator` field then create a type that omits this field (typescript can create this type).
You also have `DeploymentFlags` so you can change asa.ts method to use it and remove this method.

Reply:
do you mean tx.ts ?
I did that and used getSuggestedParams common in both, but this function can't be common as flag types are different for ASC and ASA

Reply:
and deleted asc.ts

Reply:
I know they are a little different. But you should then create a third function that has same flag processing logic and have three functions in total.
This will reduce code duplication.

Reply:
I think that would still increase the code as the code will have third type with same parameters , and will have to initialize this third with some parameters from both of them, this all for only four parameters.

I can do that if you think that is good.

Reply:
You already have that third type: `DeploymentFlags`. So you can use it here.

Reply:
Yes, understood, this is better

Reply:
Done